src/main/java/org/fcrepo/upgrade/utils/F47ToF5UpgradeManager.java

```diff
@@ -0,0 +1,185 @@
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.upgrade.utils;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.io.IOUtils;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.riot.Lang;
+import org.apache.jena.riot.RDFDataMgr;
+import org.fcrepo.client.FcrepoLink;
+
+import java.io.*;
+import java.net.URLConnection;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * @author dbernstein
+ * @since 2019-08-05
+ */
+class F47ToF5UpgradeManager extends UpgradeManagerBase implements UpgradeManager {
+
+    private static final org.slf4j.Logger logger = getLogger(F47ToF5UpgradeManager.class);
+
+
+    /**
+     * Constructor
+     *
+     * @param config the upgrade configuration
+     */
+    F47ToF5UpgradeManager(final Config config) {
+        super(config);
+    }
+
+    @Override
+    public void start() {
+        //walk the directory structure
+        processDirectory(this.config.getInputDir());
+    }
+
+    private void processDirectory(final File dir) {
+        logger.info("Processing directory: {}", dir.getAbsolutePath());
+        try (final Stream<Path> walk = Files.walk(Paths.get(this.config.getInputDir().toURI()))) {
+            //process files
+            final List<String> files = walk.filter(path -> Files.isRegularFile(path))
+                    .map(x -> x.toString()).collect(Collectors.toList());
+            files.forEach(file -> {
+                processFile(new File(file));
+            });
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void processFile(final File file) {
+        final String inputPath = this.config.getInputDir().getAbsolutePath();
+        final String absolutePath = file.getAbsolutePath();
+        final String relativePath = absolutePath.substring(inputPath.length());
+        final File newLocation = new File(this.config.getOutputDir().getAbsolutePath() + relativePath);
+        newLocation.getParentFile().mkdirs();
+        logger.debug("copy file {} to {}", file.getAbsolutePath(), newLocation.getAbsoluteFile());
+        try {
+            Files.copy(file.toPath(), new FileOutputStream(newLocation));
```

Comment:
Another option is to use `FileUtils.copyFile(file, newLocation)` which has the added bonus of internally using FileChannels, which can have performance benefits.